account_invoice_pricelist_discount/models/account_invoice.py

```diff
@@ -0,0 +1,50 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models, api
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.onchange('product_id', 'quantity', 'uom_id')
+    def _onchange_product_id_account_invoice_pricelist(self):
+
+        pricelist = self.invoice_id.pricelist_id
+
+        if not pricelist or not self.invoice_id.partner_id:
+            return
+
+        product = self.product_id.with_context(
+            lang=self.invoice_id.partner_id.lang,
+            partner=self.invoice_id.partner_id.id,
+            quantity=self.quantity,
+            date_order=self.invoice_id.date_invoice,
+            pricelist=pricelist.id,
+            uom=self.uom_id.id,
+            fiscal_position=(
+                self.invoice_id.partner_id.property_account_position_id.id)
+        )
+
+        if pricelist.discount_policy == 'without_discount':
+            if product.list_price != 0:
```

Comment:
What happens if product.list_price == 0?
If there is no action for this condition is involved then why not to check this value in the previous "if" at L28?

Reply:
fixed